MCUboot/README.md

```diff
@@ -1,163 +1,52 @@
 
-1. Extract the public key into a C data structure so that it can be built into the bootloader:
+## Demonstration
 
-   ```shell
-   mcuboot/scripts/imgtool.py getpub -k signing-keys.pem >> signing_keys.c
-   ```
-
-1. Build the bootloader:
-
-   ```shell
-   mbed compile -t GCC_ARM -m NRF52840_DK
-   ```
-
-#### Building and signing the primary application
-
-1. Build the primary application:
-
-   ```shell
-   cd ../application && mbed compile -t GCC_ARM -m NRF52840_DK
-   ```
-
-1. Copy the HEX file into the bootloader folder:
-
-   ```shell
-   cp BUILD/NRF52840_DK/GCC_ARM/application.hex ../bootloader && cd ../bootloader
-   ```
-
-1. Sign the initial application using the rsa-2048 keys:
-
-   ```shell
-   mcuboot/scripts/imgtool.py sign -k signing-keys.pem --align 4 -v 0.1.0 --header-size 4096 --pad-header -S 0xC0000 --pad application.hex signed_application.hex
-   ```
-   
-#### Creating and flashing the "factory firmware"
-
-1. Merge the bootloader and signed application:
-
-   ```shell
-   hexmerge.py -o merged.hex --no-start-addr BUILD/NRF52840_DK/GCC_ARM/bootloader.hex signed_application.hex
-   ```
-
-1. Erase the chip:
-   
-   ```shell
-   pyocd erase --chip
-   ```
-   
-1. Flash the "factory firmware" onto the board by copying `merged.hex` over to its mount point:
-
-   ```shell
-   cp merged.hex <mount_point>
-   ```
-   
-   The target's mount point can be found by running `mbedls`.
-   
-#### Creating the update binary
-
-1. Change the application's version number in its [mbed_app.json](target/application/mbed_app.json) to 0.1.1 and rebuild it:
-
-   ```shell
-   cd ../application && mbed compile -t GCC_ARM -m NRF52840_DK
-   ```
-
-1. Copy the HEX file into the bootloader folder:
-
-   ```shell
-   cp BUILD/NRF52840_DK/GCC_ARM/application.hex ../bootloader && cd ../bootloader
-   ```
-
-1. Sign the update application using the rsa-2048 keys:
-
-   ```shell
-   mcuboot/scripts/imgtool.py sign -k signing-keys.pem --align 4 -v 0.1.1 --header-size 4096 --pad-header -S 0x55000 application.hex signed_update.hex
-   ```
-
-1. Generate a raw binary file from `signed_update.hex` so that it can be transported over BLE:
-
-   ```shell
-   arm-none-eabi-objcopy -I ihex -O binary signed_update.hex signed_update.bin
-   ```
-
-### Demonstration
-
-1. Open a serial terminal on the host PC to receive serial prints from the target:
+Please make sure to activate the virtual environment, located in the root folder, before proceeding further. Also, ensure that the target board is connected and has been flashed with the factory firmware from the build stage.
 
+1. Open a serial terminal on the host computer to receive serial prints from the _FOTA target_:
    ```shell 
    mbed term -b 115200
    ```
-
-2. In a separate window, run the client script:
-
+2. In a separate shell window, run the test script from the client directory:
    ```shell
-   python client/client.py
+   python client.py
```

Comment:
why did you remove the path? the file is still in the same path, no?

Reply:
Yep, the file is in the same path. I've mentioned in the steps that the script should be run from **within the client directory**. I noticed an issue earlier when trying to run it from mock directory - it doesn't seem to work as evidenced by the screenshot below. You can see that once I `cd` into the client directory and run the script again, it works. My guess is that this has something to do with `logging.conf`. I think [Noonfom](https://github.com/noonfom) might be able to provide a bit more insight into this once he's back from holiday. 

<img width="1051" alt="CleanShot 2021-08-06 at 10 36 26@2x" src="https://user-images.githubusercontent.com/31354721/128490699-4c337fec-87f6-48f5-8f4f-2bdaad799f80.png">